src/components/task/task-button/task-button.tsx

```diff
@@ -0,0 +1,99 @@
+import React, {useCallback, useMemo} from "react";
+import Button from "@mui/material/Button";
+import Box from "@mui/material/Box";
+import {isStepOptional, STEPS} from "../task";
+import {useAppDispatch, useAppSelector} from "../../../app/hooks";
+import {
+    addSkippedStep,
+    addTask,
+    nextStep,
+    prevStep,
+    resetStep,
+    selectTasks,
+    setCustomSkipped, setTempScheduledDate, toggleTaskCreator,
+    unskipStep
+} from "../task-slice";
+import {useNavigate} from "react-router-dom";
+import dayjs from "dayjs";
+
+export const TaskButton = () => {
+    const dispatch = useAppDispatch();
+    const navigate = useNavigate();
+    const {name, description, currentStep, skipped, tempScheduledDate} = useAppSelector(selectTasks);
+
+    const isStepSkipped = (step: number) => {
+        return skipped.includes(step);
+    };
+
+    const handleBack = useCallback(() => {
+        dispatch(prevStep());
+    }, [dispatch]);
+
+    const handleSkip = useCallback(() => {
+        if (!isStepOptional(currentStep)) {
+            throw new Error("You can't skip a step that isn't optional.");
+        }
+
+        dispatch(nextStep());
```

Comment:
I assume we should not skip step (move to the next step) if this step is not optional, should not we?